### PR TITLE
[FIX] POS : security rules

### PR DESCRIPTION
**Pull request by beledouxdenis**

### CODE REVIEW

File: addons/point_of_sale/security/ir.model.access.csv [hunk 1]
```diff
@@ -1,62 +1,32 @@
 id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
 access_pos_order,pos.order,model_pos_order,group_pos_user,1,1,1,1
 access_pos_order_line,pos.order.line,model_pos_order_line,group_pos_user,1,1,1,1
-access_pos_order_manager,pos.order manager,model_pos_order,group_pos_manager,1,0,0,0
-access_pos_order_line_manager,pos.order.line manager,model_pos_order_line,group_pos_manager,1,0,0,0
 access_report_transaction_pos,report.transaction.pos,model_report_transaction_pos,group_pos_manager,1,1,1,1
 access_account_journal_pos_user,account.journal pos_user,account.model_account_journal,group_pos_user,1,0,0,0
-access_account_move_pos_user,account.move pos_user,account.model_account_move,group_pos_user,1,1,1,0
-access_account_account_pos_user,account.account pos_user,account.model_account_account,group_pos_user,1,0,0,0
-access_stock_picking_pos_user,stock.picking pos_user,stock.model_stock_picking,group_pos_user,1,1,1,1
```
Comment (beledouxdenis) [inline addons/point_of_sale/security/ir.model.access.csv]: leave it

File: addons/point_of_sale/security/ir.model.access.csv [hunk 2]
```diff
@@ -1,62 +1,32 @@
...
-access_account_analytic_account,account.analytic.account,analytic.model_account_analytic_account,group_pos_user,1,1,1,0
-access_ir_property_pos_manager,ir.property manager,base.model_ir_property,group_pos_manager,1,1,1,1
 access_account_bank_statement_line_manager,account.bank.statement.line manager,account.model_account_bank_statement_line,group_pos_manager,1,1,1,1
-access_account_invoice_manager,account.invoice manager,account.model_account_invoice,group_pos_manager,1,1,1,1
 access_product_product_supplierinfo_user,product.supplierinfo user,product.model_product_supplierinfo,group_pos_user,1,0,0,0
 access_product_product_packaging_user,product.packaging user,product.model_product_packaging,group_pos_user,1,0,0,0
-access_account_sequence_fiscalyear_manager,account.sequence.fiscalyear manager,account.model_account_sequence_fiscalyear,group_pos_manager,1,0,0,0
-access_stock_warehouse_manager,stock.warehouse manager,stock.model_stock_warehouse,group_pos_manager,1,0,0,0
-access_stock_location_manager,stock.location manager,stock.model_stock_location,group_pos_manager,1,0,0,0
-access_product_uom_manager,product.uom manager,product.model_product_uom,group_pos_manager,1,1,1,1
```
Comment (beledouxdenis) [inline addons/point_of_sale/security/ir.model.access.csv]: Leave ACLs regarding products, products uom, ...

File: addons/point_of_sale/security/point_of_sale_security.xml [hunk 3]
```diff
@@ -13,6 +13,43 @@
...
+	<field name="groups" eval="[(4, ref('group_pos_user'))]"/>
+        <field name="domain_force">[('bank_statement_id.user_id', '=', user.id)]</field>
+    </record>
+    <record id="rule_pos_cashbox_line_manager" model="ir.rule">
+        <field name="name">Point Of Sale cashbox line manager Rule</field>
+        <field name="model_id" ref="account.model_account_cashbox_line" />
+	<field name="groups" eval="[(4, ref('group_pos_manager'))]"/>
+        <field name="domain_force">[(1, '=', 1)]</field>
+    </record>
+
```
Comment (beledouxdenis) [inline addons/point_of_sale/security/point_of_sale_security.xml]: Regarding account models: Both POS/Users and POS/Managers should have the same rights: They cannot create/edit/delete bank statements if they are not accountant (except their own).
Therefore, remove all [(1, '=', 1 )] rules.

File: addons/point_of_sale/point_of_sale.py [hunk 4]
```diff
@@ -1235,7 +1236,7 @@ def action_payment(self, cr, uid, ids, context=None):
 
     def action_paid(self, cr, uid, ids, context=None):
         self.write(cr, uid, ids, {'state': 'paid'}, context=context)
-        self.create_picking(cr, uid, ids, context=context)
+        self.create_picking(cr, openerp.SUPERUSER_ID, ids, context=context)
```
Comment (beledouxdenis) [inline addons/point_of_sale/point_of_sale.py]: Leave as it is. Do not focus on rules regarding stock moves / stock pickings.

File: addons/point_of_sale/point_of_sale.py [hunk 5]
```diff
@@ -1226,7 +1227,7 @@ def insert_data(data_type, values):
             for value in group_data:
                 all_lines.append((0, 0, value),)
         if move_id: #In case no order was changed
-            self.pool.get("account.move").write(cr, uid, [move_id], {'line_id':all_lines}, context=context)
+            self.pool.get("account.move").write(cr, openerp.SUPERUSER_ID, [move_id], {'line_id':all_lines}, context=context)
```
Comment (beledouxdenis) [inline addons/point_of_sale/point_of_sale.py]: Just ensure force_company is well in the context.

File: addons/point_of_sale/point_of_sale.py [hunk 6]
```diff
@@ -1064,7 +1065,7 @@ def compute_tax(amount, tax, line):
 
             if move_id is None:
                 # Create an entry for the sale
-                move_id = account_move_obj.create(cr, uid, {
+                move_id = account_move_obj.create(cr, openerp.SUPERUSER_ID, {
```
Comment (beledouxdenis) [inline addons/point_of_sale/point_of_sale.py]: Just ensure force_company is well in the context.

File: addons/point_of_sale/point_of_sale.py [hunk 7]
```diff
@@ -969,7 +970,7 @@ def action_invoice(self, cr, uid, ids, context=None):
             inv.update(inv_ref.onchange_partner_id(cr, uid, [], 'out_invoice', order.partner_id.id)['value'])
             if not inv.get('account_id', None):
                 inv['account_id'] = acc
-            inv_id = inv_ref.create(cr, uid, inv, context=context)
+            inv_id = inv_ref.create(cr, openerp.SUPERUSER_ID, inv, context=context)
```
Comment (beledouxdenis) [inline addons/point_of_sale/point_of_sale.py]: Just ensure force_company is well in the context. Probably this isn't the case.

File: addons/point_of_sale/point_of_sale.py [hunk 8]
```diff
@@ -807,7 +808,7 @@ def create_picking(self, cr, uid, ids, context=None):
                 if line.product_id and line.product_id.type == 'service':
                     continue
 
-                move_list.append(move_obj.create(cr, uid, {
+                move_list.append(move_obj.create(cr, openerp.SUPERUSER_ID, {
```
Comment (beledouxdenis) [inline addons/point_of_sale/point_of_sale.py]: Leave picking rules as it is.

File: addons/point_of_sale/point_of_sale.py [hunk 9]
```diff
@@ -781,7 +782,7 @@ def create_picking(self, cr, uid, ids, context=None):
             picking_type = order.picking_type_id
             picking_id = False
             if picking_type:
-                picking_id = picking_obj.create(cr, uid, {
+                picking_id = picking_obj.create(cr, openerp.SUPERUSER_ID, {
```
Comment (beledouxdenis) [inline addons/point_of_sale/point_of_sale.py]: Leave picking rules as it is.

File: addons/point_of_sale/point_of_sale.py [hunk 10]
```diff
@@ -989,10 +994,10 @@ def action_invoice(self, cr, uid, ids, context=None):
                 inv_line['discount'] = line.discount
                 inv_line['name'] = inv_name
                 inv_line['invoice_line_tax_id'] = [(6, 0, [x.id for x in line.product_id.taxes_id] )]
-                inv_line_ref.create(cr, uid, inv_line, context=context)
-            inv_ref.button_reset_taxes(cr, uid, [inv_id], context=context)
+                inv_line_ref.create(cr, openerp.SUPERUSER_ID, inv_line, context=local_context)
+            inv_ref.button_reset_taxes(cr, openerp.SUPERUSER_ID, [inv_id], context=context)
```
Comment (beledouxdenis) [inline addons/point_of_sale/point_of_sale.py]: missing the use of local_context

**Comment by mart-e** [comment]
fixed in master at a29e291b69335260b26840c195599ed5bf845e16

**Pull request closed without merging**